File: app/src/main/java/de/datlag/hotdrop/util/ConnectivityUtil.java
package de.datlag.hotdrop.util;

import android.app.Activity;
import android.content.Context;
import android.net.ConnectivityManager;

import androidx.core.net.ConnectivityManagerCompat;

import java.util.Objects;

public class ConnectivityUtil {

    public static boolean isMetered(Activity activity) {
        return ConnectivityManagerCompat.isActiveNetworkMetered((ConnectivityManager) Objects.requireNonNull(activity.getSystemService(Context.CONNECTIVITY_SERVICE)));
    }
}

File: app/src/main/java/de/datlag/hotdrop/util/ConnectivityUtil.kt
package de.datlag.hotdrop.util

import android.app.Activity
import android.content.Context
import android.net.ConnectivityManager
import androidx.core.net.ConnectivityManagerCompat
import java.util.*

object ConnectivityUtil {
    fun isMetered(activity: Activity): Boolean {
        return ConnectivityManagerCompat.isActiveNetworkMetered(activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager)
    }
}